### PR TITLE
Adds support for deleting works from Orcid.

### LINKED ISSUE #1345: Support deleting ORCID work
Create a rake task that given an orcid put-code:
1. Deletes the work from ORCID.
2. Sets the put code for the contribution to null.

This is not mandated for a specific use case; however, it is inevitable that it will be necessary.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #1345

## Why was this change made?
It is our destiny to need to delete works from Orcid.

## How was this change tested?
Unit, cap-qa

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

**Comment by justinlittman** [comment]
```
bash-4.2$ RAILS_ENV=production bundler exec rake orcid:delete_work['kcasciot','275565']
Deleted work.
bash-4.2$ RAILS_ENV=production bundler exec rake orcid:delete_author_works['kcasciot']
Deleted 37 works.
```

**Review by jmartin-sul — COMMENTED** [review]
a few small suggestions, and one question for a test i didn't understand.  PR to implement the suggested changes here, since they were quick edits to make:  https://github.com/sul-dlss/sul_pub/pull/1353

since justin and i are both unavailable the rest of the week, i paired w/ @peetucket to sanity check my review this morning, and if he has time this week, he can review/merge this and the follow-on PR.

File: spec/lib/orcid/delete_works_spec.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      it 'notifies' do
+        expect(NotificationManager).to receive(:error).with(RuntimeError, /Nope!/, delete_works)
+        expect(response).to be false
+      end
+    end
+
+    context 'when Contribution already has put-code' do
+      before do
+        create :contribution, author: author, publication: publication, orcid_put_code: '1250170'
+      end
```
[the remark this answers is not in the document]
Comment (peetucket) [reply spec/lib/orcid/delete_works_spec.rb:96]: I am also not sure how this works...

[the remark this answers is not in the document]
Comment (justinlittman) [reply spec/lib/orcid/delete_works_spec.rb:96]: This test is actually using the client, which fails. That explains _why_ it is working. The real question is what it is supposed to be testing, of which I have no idea so removing test.

**Review by peetucket — APPROVED** [review]

**Pull request merged**